nflows/transforms/standard.py

```diff
@@ -1,82 +1,92 @@
 """Implementations of some standard transforms."""
 
+from typing import Optional
+import warnings
+import numpy as np
 import torch
-
+from nflows.utils.torchutils import ensure_tensor
 from nflows.transforms.base import Transform
 
 
 class IdentityTransform(Transform):
     """Transform that leaves input unchanged."""
 
-    def forward(self, inputs, context=None):
-        batch_size = inputs.shape[0]
-        logabsdet = torch.zeros(batch_size)
+    def forward(self, inputs: torch.Tensor, context=Optional[torch.Tensor]):
+        num_batches = inputs.shape[0]
+        logabsdet = torch.zeros(num_batches)
         return inputs, logabsdet
 
-    def inverse(self, inputs, context=None):
+    def inverse(self, inputs: torch.Tensor, context=Optional[torch.Tensor]):
         return self(inputs, context)
 
 
-class AffineScalarTransform(Transform):
-    """Computes X = X * scale + shift, where scale and shift are scalars, and scale is non-zero."""
-
-    def __init__(self, shift=None, scale=None):
+class PointwiseAffineTransform(Transform):
+    def __init__(
+        self,
+        shift: torch.Tensor = torch.tensor(0.0),
```

Comment:
1. Wouldn't the correct type be `Union[torch.Tensor, float]`?
2. I agree that explicit defaults (rather than `None`) are better. But my internal alarm is going off when I see a mutable default value. Can we change to scalar (hence immutable) defaults?

Reply:
Thanks for catching this, agree wholeheartedly!